libraries/tock-tbf/src/types.rs

```diff
@@ -304,6 +346,147 @@ impl core::convert::TryFrom<&[u8]> for TbfHeaderV2FixedAddresses {
     }
 }
 
+impl core::convert::TryFrom<&[u8]> for TbfHeaderDriverPermission {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderDriverPermission, Self::Error> {
+        Ok(TbfHeaderDriverPermission {
+            driver_number: u32::from_le_bytes(
+                b.get(0..4)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            offset: u32::from_le_bytes(
+                b.get(4..8)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            allowed_commands: u64::from_le_bytes(
+                b.get(8..16)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+        })
+    }
+}
+
+impl<const L: usize> core::convert::TryFrom<&[u8]> for TbfHeaderV2Permissions<L> {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderV2Permissions<L>, Self::Error> {
+        let length = u16::from_le_bytes(
+            b.get(0..2)
+                .ok_or(TbfParseError::BadTlvEntry(
+                    TbfHeaderTypes::TbfHeaderPermissions as usize,
+                ))?
+                .try_into()?,
+        );
+
+        if (length as usize) > L {
+            return Err(TbfParseError::TooManyEntries(
+                TbfHeaderTypes::TbfHeaderPermissions as usize,
+            ));
+        }
```

Comment:
This works, but requires diligent maintenance to make sure this check never changes, or the compatibility between this check and the `[]` never changes.